src/foam/dao/FixedSizeDAO.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package foam.dao;
+
+import java.util.Iterator;
+import foam.core.*;
+import foam.util.ConcurrentList;
+import foam.util.List;
+import foam.mlang.order.Comparator;
+import foam.mlang.predicate.Predicate;
+
+public class FixedSizeDAO
+  extends AbstractDAO
+{
+  private List<FObject> list;
+  public FixedSizeDAO(X x, ClassInfo classInfo, int maxSize) {
+    this(x, classInfo, maxSize, false);
+  }
+  public FixedSizeDAO(X x, ClassInfo classInfo, int maxSize, boolean isAutoRemoveOldest) {
+    setX(x);
+    setOf(classInfo);
+    list = new ConcurrentList<FObject>(maxSize, isAutoRemoveOldest);
+  }
+
+  public FObject put_(X x, FObject obj) {
+    Object key = getPrimaryKey().get(obj);
+    if ( key == null ) {
+      throw new RuntimeException("Missing Primary Key in " + this.getOf().getId() + ".put()");
+    }
+    list.enQueue(obj);
+    onPut(obj.fclone());
+    return obj;
+  }
+
+  public FObject remove_(X x, FObject obj) {
+    FObject ret = find_(x, obj);
+    if ( ret == null ) return null;
+    return list.remove(ret, 0);
+  }
+
+  public FObject find_(X x, Object o) {
+    FObject ret = null;
+    Iterator<FObject> i = list.iterator();
+    while( i.hasNext() ) {
```

Comment:
Add space between while and (. See FOAM coding style guide:
https://github.com/foam-framework/foam2/blob/master/doc/guides/StyleGuide.md